tests/extract/test_sources.py

```diff
@@ -942,6 +942,98 @@ async def r_async():
     assert async_list in allowed_results
 
 
+def test_limit_yield_cleanup() -> None:
```

Comment:
this is a good test! i had briefly thought about wether the limit influences the incremental state but I think I forgot to add a test. There are a couple of print statements remaining further down here.

Reply:
I wrote this test because I realized that we cannot limit `pg_replication` this way which is doing `yield from` iterator but full iterators MUST be consumed. So we'd need to add some kind of groupings.